### ISSUE TITLE
Coverage checker misses deinits with conditions in them

### ISSUE THREAD

**Issue by CharlesJS** [apple/swift#63852]
**Description**
If a class has a `deinit` with a conditional in it, the coverage checker will claim that the code inside the conditional has not been reached, even if it has.

**Steps to reproduce**
1. Make a Swift package via `swift package init`

2. Replace the project code with:

```swift
public class CoverageBugExample {
    let foo: Bool

    init(foo: Bool) {
        self.foo = foo
    }

    deinit {
        if self.foo {
            print("foo")
        } else {
            print("bar")
        }
    }
}
```

3. Replace the test code with:

```swift
import XCTest
@testable import CoverageBugExample

final class CoverageBugExampleTests: XCTestCase {
    func testBug() {
        do {
            _ = CoverageBugExample(foo: true)
        }

        do {
            _ = CoverageBugExample(foo: false)
        }
    }
}
```

4. Turn code coverage on, and run the tests.

5. Notice that the `deinit` method is marked as not covered, even though `foo` and `bar` both show up in the console log, proving that both cases were reached.

**Expected behavior**
If the code is reached, it should be marked as such.

**Environment**
- Swift compiler version info:

```
$ swiftc -version
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: x86_64-apple-macosx13.0
```

- Xcode version info:

```
$ xcodebuild -version
Xcode 14.2
Build version 14C18
```

- Deployment target: macOS 10.15, 11, 12, or 13. Version doesn't seem to matter.

**Comment by hamishknight** [comment]
This was fixed by #60509, you can verify with a [5.8 snapshot](https://www.swift.org/download/#snapshots) in Xcode 14.2 (unfortunately there's currently a known issue with coverage in Xcode 14.3 beta that will prevent verifying on that version)

**Comment by hamishknight** [comment]
FYI this should be resolved in Xcode 14.3 beta 2
